src/Microsoft.AspNet.OData.Shared/Query/Expressions/SelectExpandBinder.cs

```diff
@@ -1057,23 +1045,26 @@ private Expression AddOrderByQueryForSource(Expression source, OrderByClause ord
                     _settings.EnableConstantParameterization);
             }
 
-            if (_settings.PageSize.HasValue || modelBoundPageSize.HasValue || hasTopValue || hasSkipvalue)
+            if (structuredType is IEdmEntityType)
             {
-                // don't page nested collections if EnableCorrelatedSubqueryBuffering is enabled
-                if (!_settings.EnableCorrelatedSubqueryBuffering)
+                if (_settings.PageSize.HasValue || modelBoundPageSize.HasValue)
```

Comment:
Why don't we care about to and skip anymore?

Reply:
@habbes `PageSize` and `ModelBoundPageSize` apply to server-side paging. `$top` and `$top` in the manner they're used here apply to client-side paging. We want to still allow the client to do something like this:
```
EntitySet?$select=PrimitiveOrEnumOrComplexCollectionProperty($top=2)
EntitySet?$select=PrimitiveOrEnumOrComplexCollectionProperty($skip=3;$top=2)
```